### DISCUSSION TITLE
use substrate?

### DISCUSSION THREAD

**Discussion by gogo2464** [BOINC/boinc#5509] [General]
**Describe the problem**
avoid to recode everything

**Describe the solution you'd like**
We might use substrate library.I need your opinion.

**Comment by AenBleidd** [comment]
@gogo2464 
Could you please share more details? What is substrate?

**Comment by gogo2464** [comment]
it is a library to create blockchain with not having to reinvent the whell https://github.com/paritytech/polkadot-sdk/tree/master/substrate particulary nice for networking. as it use rollup, it does not require to push block on existing blockchain chains.

does not seems to exists in C/C++ but uniffy-rs-cpp could convert rust to C++.

I hope it could help

**Comment by AenBleidd** [comment]
BOINC doesn't use Blockchain

**Comment by gogo2464** [comment]
ah sorry!!!it is a serverless proof ok sorry!!!

**Comment by gogo2464** [comment]
ok because from what I see in the project, you proove that some calculus has been done.

It is a great amount of process. And I think we should use useful work to slow operation in bitcoin like crypto transaction instead of bruteforcing useless hash.

**Comment by AenBleidd** [comment]
@gogo2464, https://gridcoin.us/

**Comment by gogo2464** [comment]
thanks!

**Comment by gogo2464** [comment]
ah ok so BOINC is the list of project where gridcoin is the proof?

thanks!!!

**Comment by kiska3** [comment]
What is this discussion asking, so far I haven't deciphered its meaning yet
